engine/src/witness/btc/deposits.rs

```diff
@@ -80,14 +85,23 @@ impl<Inner: ChunkedByVault> ChunkedByVaultBuilder<Inner> {
 					})
 				};
 
-				for (broker_id, vault_address) in vault_addresses {
+				for (broker_id, channel_id, vault_address) in vault_addresses {
 					for tx in &txs {
-						if let Some(call) = super::vault_swaps::try_extract_vault_swap_call(
+						if let Some(deposit) = super::vault_swaps::try_extract_vault_swap_witness(
 							tx,
 							&vault_address,
+							channel_id,
 							&broker_id,
 						) {
-							process_call(call.into(), epoch.index).await;
+							process_call(
+								BtcIngressEgressCall::vault_swap_request {
+									block_height: header.index,
+									deposits: vec![deposit],
```

Comment:
Yeah, originally @dandanlen and I talked about this as something we could do, but would be outside of scope of this PR, but now that I have changed the extrinsic to take a Vec, I suppose we might as well batch them.

Reply:
Update: as mentioned on discord (https://discord.com/channels/775961728608895008/1311562365828857947/1311939330188644413), I changed the signature to take a single deposit rather than a Vec for now as doing otherwise was causing some weird bouncer behaviour.